sdk/src/simple_state/context.rs

```diff
@@ -0,0 +1,396 @@
+// Copyright 2019 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::marker::PhantomData;
+
+use crate::protocol::simple_state::{
+    StateEntry, StateEntryBuilder, StateEntryList, StateEntryListBuilder, StateEntryValue,
+    StateEntryValueBuilder, ValueType,
+};
+use crate::protos::{FromBytes, IntoBytes};
+use crate::simple_state::addresser::Addresser;
+use crate::simple_state::error::SimpleStateError;
+use crate::TransactionContext;
+
+/// KeyValueTransactionContext used to implement a simplified state consisting
+/// of natural keys and a ValueType, an enum object used to represent a range primitive data types.
+/// Uses an implementation of the Addresser trait to calculate radix addresses to be stored in the
+/// KeyValueTransactionContext's internal transaction context.
+pub struct KeyValueTransactionContext<'a, A, K>
+where
+    A: Addresser<K>,
+{
+    context: &'a mut dyn TransactionContext,
+    addresser: A,
+    /// PhantomData<K> is necessary for the K generic to be used with the Addresser trait, as K is not
+    /// used in any other elements of the KeyValueTransactionContext struct.
+    _key: PhantomData<K>,
+}
+
+impl<'a, A, K> KeyValueTransactionContext<'a, A, K>
+where
+    A: Addresser<K>,
+{
+    /// Creates a new KeyValueTransactionContext
+    /// Implementations of the TransactionContext trait and Addresser trait must be provided.
+    pub fn new(
+        context: &'a mut dyn TransactionContext,
+        addresser: A,
+    ) -> KeyValueTransactionContext<'a, A, K> {
+        KeyValueTransactionContext {
+            context,
+            addresser,
+            _key: PhantomData,
+        }
+    }
+
+    /// Calculates the address using the internal addresser then creates and serializes a
+    /// StateEntryList protobuf message to be stored in the internal transaction context as bytes.
+    ///
+    /// Returns an `Ok(())` if the entry is successfully stored.
+    ///
+    /// # Arguments
+    ///
+    /// * `key` - The natural key to set the state entry at
+    /// * `values` - The HashMap to be stored in state at the provided natural key
+    pub fn set_state_entry(
+        &self,
+        key: &K,
+        values: HashMap<String, ValueType>,
+    ) -> Result<(), SimpleStateError> {
+        let addr = self.addresser.compute(key.to_owned())?;
```

Comment:
Can you just call the `set_state_entries` function?  Same with the other single entry methods.